src/librbd/ExclusiveLock.cc

```diff
@@ -40,9 +40,12 @@ bool ExclusiveLock<I>::accept_requests(int *ret_val) const {
   Mutex::Locker locker(ML<I>::m_lock);
 
   bool accept_requests = (!ML<I>::is_state_shutdown() &&
+                          !ML<I>::is_state_pre_releasing() &&
                           ML<I>::is_state_locked() &&
```

Comment:
```is_state_locked``` covers several states where the actual lock is held on the object -- the ```STATE_PRE_RELEASING``` being one of those states. However, if we are in the process of releasing the lock, we shouldn't be accepting new requests nor accepting new IO.